CONTRIBUTING.md

```diff
@@ -10,12 +10,119 @@ We actively welcome your pull requests.
+3. From the CLI, open your freshly cloned Rapid directory by running the command `code Rapid`
+4. Once in VSCode, type **Ctrl+Shift+`** to open a terminal. Alternatively, use the **Terminal>New Terminal** menu option.
+5. Make sure you are using the latest `node` by running `nvm install node`
+6. Install all required packages with `npm install`
+7. Then run `npm run all` to set up Rapid.
+8. Finally, run `npm run quickstart` to start the server on port 8080.
+
+Congrats! 🎉 You should now be able to use rapid by navigating to [http://localhost:8080/](http://localhost:8080/) on Chrome.
+
+### Setting up the VSCode debugger
+VSCode provides a debugging mode using Chrome. To use it, do this:
+
+1. Go to **Run>Add Configuration** and select `{} Chrome: Launch`
+2. This will create `launch.json`.
+3. Make sure to add the `/dist/` path to `webRoot` so that it looks like this:
+```json
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}/dist/"
+        }
+    ]
+}
+```
+
+Great! You should now be able to run the Rapid server with `npm run quickstart` and then the debugger by either pressing **F5** on your keyboard, or clicking on **Run>Start Debugging**.
+
+This will launch Chrome on whatever address you provided to `url` on `launch.json`, allowing you to use Rapid as well as entering debugging mode on VSCode.
+
+> **Note:** Again, you are welcome to try your own debugging method, such as [Chrome DevTools](https://developer.chrome.com/docs/devtools/javascript/breakpoints/).
+
+### Debugging example
+
+When hacking Rapid, it will be useful to watch the code execute line-by-line. To do this, you can set up inline breakpoints. You can learn all about it in [the official documentation](https://code.visualstudio.com/docs/editor/debugging).
+
+A good place to start is the `/index.html` file. Set a breakpoint on the `context.initAsync();` call from it's last few lines:
+
+```javascript
+...
+          window.context = context;  // for debugging
+          context.initAsync();
+        }
+      }
+    </script>
+  </body>
+</html>
+```
+Then run `npm run quickstart` and press **F5** to launch Chrome along with the VSCode debugger.
+
+This will freeze execution right at the moment of Rapid initialization. Click on the **Step in** button on the debugger controls (the arrow pointing down into a point) to enter the `context.initAsync()` function. Then click on **Step over** repeatedly to watch the code execute line-by-line.
```

Comment:
Might be nice to add some arrow emojis to help visualize the debugger buttons -> [emoji-cheat-sheet](https://github.com/ikatyang/emoji-cheat-sheet/blob/master/README.md#arrow )